lib/connect-backbone-to-react.js

```diff
@@ -0,0 +1,147 @@
+const hoistStatics = require('hoist-non-react-statics');
+const { Component, PropTypes, createElement } = require('react');
+const debounceFn = require('lodash.debounce');
+const isArray = require('lodash.isarray');
+
+function getDisplayName(name) {
+  return `connectBackboneToReact(${name})`;
+}
+
+function defaultMapModelsToProps(models) {
+  return Object.keys(models).reduce((acc, modelKey) => {
+    const model = models[modelKey];
+    acc[modelKey] = model.toJSON();
+    return acc;
+  }, {});
+}
+
+module.exports = function connectBackboneToReact(
+  mapModelsToProps,
+  options = {}
+) {
+  if (typeof mapModelsToProps !== 'function') {
+    mapModelsToProps = defaultMapModelsToProps;
+  }
+
+  const {
+    debounce = false,
+    events = {},
+    modelTypes = {},
+  } = options;
+
+  function getEventNames(modelName) {
+    let eventNames = events[modelName];
+
+    // Allow turning off event handlers by setting events to false.
+    if (eventNames === false) {
+      return [];
+    }
+
+    if (!isArray(eventNames)) {
+      eventNames = [];
+    }
+
+    if (eventNames.length === 0) {
+      eventNames.push('all');
+    }
+
+    return eventNames;
+  }
+
+  function validateModelTypes(modelsMap) {
+    return Object.keys(modelTypes).forEach(modelKey => {
+      const ModelConstructor = modelTypes[modelKey];
+      const modelInstance = modelsMap[modelKey];
+
+      const isInstanceOfModel = modelInstance instanceof ModelConstructor;
+      if (!isInstanceOfModel) {
+        throw new Error(`"${modelKey}" model found on modelsMap does not match type required.`);
+      }
+    });
+  }
+
+  return function createWrapper(WrappedComponent) {
+    const wrappedComponentName = WrappedComponent.displayName
+      || WrappedComponent.name
+      || 'Component';
+
+    const displayName = getDisplayName(wrappedComponentName);
+
+    class Connect extends Component {
+      constructor(props, context) {
+        super(props, context);
+
+        this.modelsMap = props.models || context.models;
```

Comment:
it's a bit confusing having `modelsMap` and `mapModelsToProps`. Could you just have `this.models` instead?